src/index.ts

```diff
@@ -1,13 +1,86 @@
 import * as ts from "typescript";
-import { PluginImpl } from "rollup";
+import * as path from "path";
+import { PluginImpl, SourceDescription, InputOptions } from "rollup";
 import { Transformer } from "./Transformer";
 import { NamespaceFixer } from "./NamespaceFixer";
 
+const dts = ".d.ts";
+const tsx = /\.tsx?$/;
+
+const formatHost: ts.FormatDiagnosticsHost = {
+  getCurrentDirectory: () => ts.sys.getCurrentDirectory(),
+  getNewLine: () => ts.sys.newLine,
+  getCanonicalFileName: ts.sys.useCaseSensitiveFileNames ? f => f : f => f.toLowerCase(),
+};
+
+const getCompilerOptions = (input: string): ts.CompilerOptions => {
+  const configPath = ts.findConfigFile(path.dirname(input), ts.sys.fileExists);
+  if (!configPath) {
+    return {};
+  }
+  const { config, error } = ts.readConfigFile(configPath, ts.sys.readFile);
+  if (error) {
+    console.error(ts.formatDiagnostic(error, formatHost));
+    return {};
+  }
+  const { options, errors } = ts.parseJsonConfigFileContent(config, ts.sys, path.dirname(configPath));
+  if (errors.length) {
+    console.error(ts.formatDiagnostics(errors, formatHost));
+    return {};
+  }
+  return options;
+};
+
+const createProgram = ({ input }: InputOptions) => {
+  if (typeof input !== "string") {
+    throw new TypeError('"input" option must be a string');
```

Comment:
well this fails the multiple entrypoints test

Reply:
I will fix by calling `ts.createProgram` once per entry point. Sound good?

Reply:
We could add a way to configure the plugin to use a single `ts.Program` for all the entry points, as long as they're all `.ts` modules (and not `.d.ts` modules). It just needs to know which `tsconfig.json` to use.